src/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -795,21 +796,36 @@ public IEnumerable<string> GetFiles(string path, string filter, bool recursive)
                         select p.Name;
                 });
         }
-
+        
         public IEnumerable<string> GetFullPaths(string fileName)
         {
             return ThreadHelper.JoinableTaskFactory.Run(async delegate
             {
                 await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                MicrosoftBuildEvaluationProject buildProject = EnvDTEProjectUtility.AsMicrosoftBuildEvaluationProject(EnvDTEProject);
-                return buildProject.Items.Where(
-                    projectItem =>
+                var paths = new List<string>();
+                var q = new Queue<EnvDTEProjectItems>();
+                q.Enqueue(EnvDTEProject.ProjectItems);
+                while (q.Count > 0)
+                {
+                    var items = q.Dequeue();
+                    foreach (EnvDTE.ProjectItem item in items)
                     {
-                        var itemFileName = Path.GetFileName(projectItem.EvaluatedInclude);
-                        return string.Equals(fileName, itemFileName, StringComparison.OrdinalIgnoreCase);
-                    })
-                    .Select(projectItem => Path.Combine(ProjectFullPath, projectItem.EvaluatedInclude));
+                        if (item.Kind == NuGetVSConstants.VsProjectItemKindPhysicalFile)
+                        {
+                            if (StringComparer.OrdinalIgnoreCase.Equals(item.Name, fileName))
```

Comment:
Does this name include the path? It seems like this could hit the same issue we have with projects where UniqueName is needed.